src/Workspaces/Core/Portable/Options/ILegacyWorkspaceOptionService.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.Options;
+
+internal interface ILegacyWorkspaceOptionService : IWorkspaceService
```

Comment:
for my education - if there is code currently using `GetOption` / `SetOptions` here, what is the non-legacy way to do it?

Reply:
I'll add a comment. This service is only used to facilitate reading/writing global options from Workspace and Solution APIs that are obsolete. Global options should be accessed through IGlobalOptionService.